### DISCUSSION TITLE
Searchengine replacement

### DISCUSSION THREAD

**Discussion by PPetterssonSO** [SuperOffice/typescript-for-superoffice#20] [General]
May we please have an example of the equivalent function of an SearchEngine,
Including the following parameters for AddCriteria:

String compOperator, String field2, String func2, String rowOperator, Integer priority

**Comment by ejfasting** [comment]
Hello @Nawksen, 

I believe the [dynamic archive](https://docs.superoffice.com/en/api/netserver/search/odata/dynamic-provider.html) is the best replacement for the SearchEngine, if you want to make 'complicated' queries for specific data. It lets you create the same kind of query as you would with the searchengine, and follows the same `dot syntax`.

For example, using the [GetArchiveListByColumns2Async](https://docs.superoffice.com/en/api/reference/webapi/SuperOffice.WebApi.Agents.IArchiveAgent.html#SuperOffice_WebApi_Agents_IArchiveAgent_GetArchiveListByColumns2Async_System_String_System_String_System_String_System_String_System_String_System_Int32_System_Int32_SuperOffice_WebApi_RequestOptions_):

```typescript
const agent = soApi.getArchiveAgent();
var result = await agent.getArchiveListByColumns2Async(
                "dynamic",
                "contact.contact_id,contact.name,contact.category_idx",
                "contact.name",
                "contact.contact_id eq 5",
                "",
                0,
                10
            );

const obj = JSON.stringify(result, null, 2);
context.result.body = obj;
```

Please note that in this spesific example I'm getting a contact based on contact_id eq 5, and using the dynamic archive for this is a little overkill as I could use the [SimpleContact Archive](https://docs.superoffice.com/en/api/netserver/archive-providers/reference/simplecontact.html) instead..

Does this answer your question? :)

PS: I'll add this as an example to the repo
//Eivind
